utils/misc.py

```diff
@@ -43,3 +44,65 @@ def print_progress_bar(iteration, total, prefix= '', suffix='', decimals=1, leng
     # Print New Line on Complete
     if iteration == total:
         print()
+
+
+def preprocess_var_refs(root, base_vars=None, remove_vars_dict=True):
+    """
+    Processes the dict by replacing var refs with variables declared in the 'vars' dict.
```

Comment:
These comments are not descriptive enough, maybe add a simple example (one of the unit tests will do).